src/llm/model.py

```diff
@@ -0,0 +1,285 @@
+from typing import List
+import torch
+import math
+from torch import Tensor
+from torch import nn
+from dataclasses import dataclass
+from jaxtyping import Float32
+
+
+@dataclass
+class TransformerConfig:
+    model_dim: int
+    expansion_dim: int
+    num_heads: int
+    num_layers: int
+
+
+class FeedForward(nn.Module):
+    def __init__(
+        self,
+        model_dim: int,
+        expansion_dim: int,
+    ):
+        super().__init__()
+        self.layer1 = nn.Linear(model_dim, expansion_dim, bias=True)
+        self.activation = nn.ReLU()
+        self.layer2 = nn.Linear(expansion_dim, model_dim, bias=True)
+
+    def forward(
+        self,
+        inputs: Float32[Tensor, "batch seq_len model_dim"],
+    ) -> Float32[Tensor, "batch seq_len model_dim"]:
+        x = self.layer1(inputs)
+        x = self.activation(x)
+        x = self.layer2(x)
+        return x
+
+
+class ScaledDotProductAttention(nn.Module):
+    def __init__(
+        self,
+        model_dim: int,
+        dim_k: int,
+        dim_v: int,
+    ):
+        super().__init__()
+        self.W_query = nn.Linear(model_dim, dim_k, bias=False)
+        self.W_key = nn.Linear(model_dim, dim_k, bias=False)
+        self.W_value = nn.Linear(model_dim, dim_v, bias=False)
+        self.dim_k = dim_k
+
+    def forward(
+        self,
+        query: Float32[Tensor, "batch seq_len model_dim"],
```

Comment:
the type should be `Float` instead of `Float32`. 

from docs:

"""
Any floating point: Float
Of particular precision: BFloat16, Float16, Float32, Float64
"""

Reply:
Let's change it everywhere.